secure_message/v2/repository/retriever.py

```diff
@@ -33,6 +33,24 @@ def unread_message_count_by_survey(user, survey):
             raise InternalServerError(description="Error retrieving count of unread messages from database")
         return result
 
+    @staticmethod
+    def message_count_by_survey(user, survey):
+        """Count users messages for a specific survey"""
+        if user.is_internal:
+            status_conditions, survey_conditions = RetrieverV2._get_conditions_internal_user(survey, user)
+        else:
+            status_conditions, survey_conditions = RetrieverV2._get_conditions_respondent(survey, user)
+
+        try:
+            result = SecureMessage.query.join(Status).\
+                filter(or_(*status_conditions)).\
+                filter(and_(*survey_conditions)).\
+                distinct(SecureMessage.thread_id).count()
+        except Exception as e:
+            logger.error('Error retrieving count of messages by survey from database', error=e)
+            raise InternalServerError(description="Error retrieving count of unread messages from database")
+        return result
```

Comment:
This method shares a lot of code with `unread_message_count_by_survey`, can we rationalise into one method where label is a parameter?